woocommerce/woocommerce
# [Experimental] Interactivity Dropdown multi-select mode, ratings filter and introduce each directive

### Changes proposed in this Pull Request:

1. Introduce a multi-select mode for the interactivity dropdown component.
2. Thanks to @DAreRodz contribution, introduces the interactivity each directive and fixes for context sync issues
3. Integrate the multi-select dropdown mode to the recently introduced collection rating filter block





Closes #42053 #42052



### How to test the changes in this Pull Request:

1. Insert a product collection (beta) block and the new "collection filters" block
2. Note that the ratings filter is automatically inserted to collection filters as part of the template
3. Open inspector for the rating filter block. Choose "dropdown" as display style.
4. Choose "multiple" as select type
5. Update and view in front-end. Test adding/removing filters from the dropdown
6. Go back to editor and choose "single" as select type
7. update and view in front end and check the single style still works.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 
Add experimental multi-select mode for interactivity dropdown component of blocks.

</details>
